### PR TITLE
Add dev-listaddrs command

### PR DESCRIPTION
**Pull request by lvaccaro**
Add dev-listaddrs command to show bip32 addresses list (up to bip32_max_index).
#### help

``` shell
$ lightning-cli help
...
command=dev-listaddrs
description=Show addresses list up to derivation {index} (default is the last bip32 index)
...
```
#### output
``` shell
$ lightning-cli dev-listaddrs
{ "addresses" : 
        [ 
                { 
                         "keyidx" : 0, 
                         "address" : "...", 
                         "pubkey" : "...", 
                         "redeemscript" : "..." 
                }, 
                .... 
        ]
}
```
Set optional value {index} to show also future derivations up to {index}. This could be useful for testing and recovery.

### CODE REVIEW

File: wallet/walletrpc.c [hunk 1]
```diff
@@ -395,6 +396,90 @@ static const struct json_command newaddr_command = {
...
+	const tal_t *tmpctx = tal_tmpctx(NULL);
+
+    if (!json_get_params(cmd, buffer, params,
+                         "?bip32_max_index", &bip32tok, NULL)) {
+		tal_free(tmpctx);
+        return;
+    }
+
+	if (!bip32tok || !json_tok_u64(buffer, bip32tok, &bip32_max_index)) {
+        bip32_max_index = db_get_intvar(cmd->ld->wallet->db, "bip32_max_index", 0);
```
Comment (cdecker) [inline wallet/walletrpc.c]: Minor nit: inconsistent use of spaces vs. tabs

File: wallet/walletrpc.c [hunk 2]
```diff
@@ -395,6 +396,90 @@ static const struct json_command newaddr_command = {
...
+
+		json_object_start(response, NULL);
+		json_add_u64(response, "keyidx", keyidx);
+		json_add_string(response, "address",
+						p2sh_to_base58(cmd, get_chainparams(cmd->ld)->testnet,
+									   &p2sh));
+
+		{
+			char *hex = pubkey_to_hexstr(tmpctx, &pubkey);
+			json_add_string(response, "pubkey", hex);
```
[the remark this answers is not in the document]
Comment (lvaccaro) [reply wallet/walletrpc.c]: Do you think it is better to show both bech32 & p2sh address for the same derivation like that:
```
$ lightning-cli dev-listaddrs
{ "addresses" : [
{                                                                                                                                              
"keyidx": 0,                                                                                                                                 
"pubkey": "...",      
"p2sh": "...",   
"p2sh_redeemscript": "...",
"bech32": "...",
"bech32_redeemscript": "..." 
} ] }
```
or use address_type option (like get_newaddr) to show only one kind of derivation address?
```
$ lightning-cli dev-listaddrs bech32
{ "addresses" : [
{ 
"keyidx" : 0, 
"address" : "...", 
"pubkey" : "...", 
"redeemscript" : "..." 
} ] }
```

**Comment by cdecker** [comment]
Thanks for updating this @lvaccaro, if you enable edits by owners when creating a PR you'll allow us to amend and add fixes to your PR btw.

Please don't create merge requests, but rather rebase on top of `master` since that makes reviewing a lot simpler.

**Comment by lvaccaro** [comment]
Sorry, I close this pr and I will open a new pr from master.

**Pull request closed without merging**